source/common/stream_info/filter_state_impl.cc

```diff
@@ -152,5 +167,44 @@ void FilterStateImpl::maybeCreateParent(ParentAccessMode parent_access_mode) {
   parent_ = lazy_create_ancestor.first;
 }
 
+absl::Status FilterStateImpl::addAncestor(const FilterStateSharedPtr& ancestor) {
+  if (!ancestor) {
+    return absl::OkStatus();
+  }
+  if (ancestor->lifeSpan() <= life_span_) {
+    return absl::FailedPreconditionError(absl::StrCat("Ancestor lifespan ", ancestor->lifeSpan(),
+                                                      " must be larger than current lifespan ",
+                                                      life_span_));
+  }
+  if (parent_) {
```

Comment:
So this means that if upstream request stream info has set data at connection life span, it'll fail the precondition above? can we make it a bit more clear since the error message will be confusing.

Reply:
That's correct. I updated errors for when it's a parent that has the conflict.